### DISCUSSION TITLE
More examples

### DISCUSSION THREAD

**Discussion by totaland** [uNetworking/uWebSockets#1357] [Q&A]
Hi guys, 
I am struggling to learn this library and with the concept of socket API. I have read about socket API in MDN, do a few examples for chatting with socket.io but I still can't make uWebSockets to work when I apply it in NextJs project. 

I was wondering if you have some repos that are completed frontend with Websocket API and backend with uWebSocket that you are willing to share with a newbie like me to learn about this library?

Much appreciate your time 🙏🏻

**Comment by frankhale** [comment]
@totaland This question is now quite old so I was wondering if you found any information out yet? I am also struggling to learn this library. The examples fall off pretty quickly once you get hello,world working. I'm not really finding any info on Google pertaining to assisting in learning this. I've been looking at the code and trying to figure it out that way.

**Reply by totaland** [reply]
Hey I figured it out that once you have the backend running. All you want to do is establish the connection to the backend 
 in the useEffect.

`useEffect(() => {
const socket = new Websocket(‘ws://localhost:9001’)
socket.onopen = () => {do something}
socket.onmessage = (e) => {do other thing}
etc
}, [])`

**Reply by frankhale** [reply]
Oh cool. Thanks for that. I'm still working on getting the backend squared away.
